File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
    static void swap(vector<int>& arr,int l,int r){
        while(l<r){
            int t=arr[l];
            arr[l]=arr[r];
            arr[r]=t;
            l++;
            r--;
        }
    }
    void rotate(vector<int>& nums, int k) {
        k%=nums.size();
        swap(nums,0,nums.size()-1);
        swap(nums,0,k-1);
        swap(nums,k,nums.size()-1);
    }
};

File: 189-rotate-array/rotate-array.java
class Solution {
    private static void swap(int nums[],int l,int r){
        while(l<r){
            int t=nums[l];
            nums[l]=nums[r];
            nums[r]=t;
            l++;
            r--;
        }
    }
    public void rotate(int[] nums, int k) {
        k%=nums.length;
        swap(nums,0,nums.length-1);
        swap(nums,0,k-1);
        swap(nums,k,nums.length-1);
    }
    
}